CIP 4 and 5, Module Readiness Checklist and Committee (#6)
cosmos/cips: CIPS/cip-4.md

---
cip: 4 
title: Module Readiness Checklist 
author: Shahan Khatchadourian <@shahankhatch>
status: Living 
type: Meta 
created: 2021-05-05
---

This document is used as a template for Module Readiness Checklists.

It is intended to serve mainnet decision makers with relevant information for deciding when a module is ready for deployment onto a mainnet, e.g., Cosmos Hub.

The checklist is intended to be easy to complete by the Module Readiness Committee.

> Not all items on the checklist need to have items.

Reviewer: ```suggestion
Not all items on the checklist apply to every module, they should be removed based on context.
```

Is this what is meant?

Author: Yes, thanks.

# Module Readiness Checklist: `Module Name`

`Summary`

Overview

- [ ] ADR
- [ ] Specification
- [ ] Documentation

> - [ ] Light Paper

Reviewer 2: ```suggestion
- [ ] Light paper
```

- [ ] Repository
- [ ] Demo

Continuity

- [ ] Developer:
- [ ] Maintainer:
- [ ] Handoff requirements:
- [ ] Support and training requirements:

Integrations

- [ ] Gaia integration
- [ ] Integration partners:

> - [ ] IBC Readiness

Reviewer 2: ```suggestion
- [ ] IBC readiness
```

- [ ] Downstream user impact report
- [ ] Upstream partner impact report

> - [ ] Inter-module dependence, Ethereum Full node, any potential NFT module on the Hub

Reviewer 2: ```suggestion
- [ ] Inter-module dependence, Ethereum full node, any potential NFT module on the Hub
```

Governance

- [ ] Hub governance proposal
- [ ] Core community governance

> - [ ] Steering community

Reviewer 2: ```suggestion
- [ ] Steering committee
```

> - [ ] Timelines and Roadmap

Reviewer 2: ```suggestion
- [ ] Timeline and roadmap
```

Research

> - [ ] Working Group/Interest Group and Charter

Reviewer 2: ```suggestion
- [ ] Working group/interest group and charter
```
